src/cloudai/_core/registry.py

```
@@ -203,6 +205,52 @@ def update_agent(self, name: str, value: Type[BaseAgent]) -> None:
+            raise KeyError(f"Agent '{name}' not found. Available agents: {self.agent_names()}")
+
+        ep = self.agent_entrypoints_map[name]
+        cls = ep.load()
+
+        from cloudai.configurator.base_agent import BaseAgent
+
+        if not issubclass(cls, BaseAgent):
+            warnings.warn(
+                f"Skipping entrypoint: {name} -> {ep.value} class={cls} (not a subclass of BaseAgent)", stacklevel=2
+            )
+            raise TypeError(f"Entry point agent '{name}' resolved to {cls}, which is not a subclass of BaseAgent.")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)